logging/bin/getlogs.py

```diff
@@ -95,6 +103,21 @@ def validate_input(dict):
             print("Please remove single quotes ('') from search argument.")
             sys.exit()
 
+def open_port(dict):
+    """Binds the v4m-search service on port 9200 to a locally available port"""
+    #Get open port
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(("",0))
+    s.listen(1)
+    port = s.getsockname()[1]
```

Comment:
Does this reserve/lock the port preventing anything else from using it?  If not, I suspect there is a very small chance that the port identified here is taken by some other process before the kubectl command executes.  Probably not worth worrying about but a potential issue.

The kubectl port-forward command will grab a random port if none is provided but then you have to figure out which port was used...that may be difficult/messy.

Reply:
The `bind` command above will bind the socket to any open port which should take the risk you mention out of question. Once the socket binds it should own the port